google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/FieldValues.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery;
+
+import com.google.common.collect.ImmutableList;
+import java.io.Serializable;
+import java.util.AbstractList;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Google BigQuery Table Field Values class, which represents a row in returned query result
+ * (table row). Tables rows can be retrieved as a result of a query or when listing table data.
+ *
+ * Depending on how a corresponding query was executed, each row (an instance of {@code FieldValues})
+ * may or may not contain related schema. If schema is not provided, the individual cells of the row
+ * will still be accessible by index but not by name.
+ */
+public class FieldValues extends AbstractList<FieldValue> implements Serializable {
```

Comment:
It seems like this should be called `FieldValueList`

Reply:
I like FieldValues and Fields more, as they are shorter and still clear. But I changed to FieldList and FieldValueList to make clear that those are valid lists.